core/include/sched.h

```diff
@@ -278,6 +278,37 @@ typedef void (*sched_callback_t)(kernel_pid_t active, kernel_pid_t next);
 void sched_register_cb(sched_callback_t callback);
 #endif /* MODULE_SCHED_CB */
 
+#if IS_USED(MODULE_SCHED_RUNQUEUE_API) || defined(DOXYGEN)
+/**
+ * @brief   Scheduler runqueue callback
+ *
+ * @param   prio      the priority of the runqueue that changed
+ */
+typedef void (*sched_runqueue_callback_t)(uint8_t prio);
+
+/**
+ * @brief  Register a callback that will be called on every runqueue change
+ *
+ * @param[in] callback The callback functions that will be called
+ */
+void sched_register_runqueue_cb(sched_runqueue_callback_t callback);
```

Comment:
If i do that the decision whether to implement the calling of that function or not is moved to LTO i think, since until the no part of the compile chain knows if it is implemented anywhere

Reply:
This is not related to LTO. LTO will (hopefully!) also fail to link if an undefined symbol is called, rather than silently optimizing the call out.

What I had in mind is:

```C
    if (IS_USED(MODULE_SCHED_RUNQ_CALLBACK)) {
        extern void sched_runq_callback(uint8_t prio);
        sched_runq_callback(prio);
    }
```

Reply:
having the linker fail on undefined is possible i will add a commit that does this, but on the first glance this seems to me like a call from core to sys.

Reply:
> having the linker fail on undefined is possible i will add a commit that does this

I'm pretty certain that the linker will fail when referencing undefined symbols, unless someone explicitly disables this. And I'm pretty sure that this is not done in RIOT - so linking should fail when referencing undefined symbols. If this is not the case, it would be a bug.